utilities/pulumi_state_splitter/tests/test_split.py

```diff
@@ -0,0 +1,365 @@
+            backend_dir=self._tmp_dir,
+            stack_name=data.MULTI_STACK_NAMES[0],
+        )
+        state_file.load()
+        pulumi_state_splitter.split.StateDir.split_state_file(state_file)
+        got = util.Directory.load(self._tmp_dir)
+        want = input_
+
+        # only one of the stacks should be processed
+        want[".pulumi"]["stacks"]["test-project-1"].pop("test-stack-1.json")
+        want["test-project-1"] = {
+            "test-stack-1": data.multi_stack_split()["test-project-1"]["test-stack-1"]
+        }
+        want.compare(got, self)
+
+    def test_unsplit_state(self):
+        """Testing `StateDir.unsplit`."""
+        input_ = data.multi_stack_split()
+        input_.save(self._tmp_dir)
+        state_dir = pulumi_state_splitter.split.StateDir(
+            backend_dir=self._tmp_dir,
+            stack_name=data.MULTI_STACK_NAMES[0],
+        )
+        state_dir.load()
+        state_dir.unsplit()
+        got = util.Directory.load(self._tmp_dir)
+        want = data.multi_stack_unsplit()
+        want = input_
+        # only one of the stacks should be processed
+        want["test-project-1"].pop("test-stack-1")
+        want[".pulumi"] = {
+            "stacks": {
+                "test-project-1": {
+                    "test-stack-1.json": data.multi_stack_unsplit()[".pulumi"][
+                        "stacks"
+                    ]["test-project-1"]["test-stack-1.json"]
+                }
+            }
+        }
+        want.compare(got, self)
+
+
+class TestUnsplitter(util.TmpDirTest):
+    """Testing `Unsplitter`."""
+
+    def test_unsplitter_some_stacks(self):
+        """Testing `Unsplitter` with specified stacks."""
+        input_ = data.multi_stack_split()
+        input_.save(self._tmp_dir)
+
+        with pulumi_state_splitter.split.Unsplitter(
+            all_stacks=False,
+            backend_dir=self._tmp_dir,
+            stacks_names=data.MULTI_STACK_NAMES[:2],
+        ):
+            got = util.Directory.load(self._tmp_dir)
+            want = data.multi_stack_unsplit()
+            want[".pulumi"]["stacks"].pop("test-project-2")
+            want["test-project-2"] = input_["test-project-2"]
+            want.compare(got, self)
+
+        got = util.Directory.load(self._tmp_dir)
+        input_.compare(got, self)
+
+    def test_unsplitter_all_stacks(self):
+        """Testing `Unsplitter` with specified stacks."""
+        input_ = data.multi_stack_split()
+        input_.save(self._tmp_dir)
+
+        with pulumi_state_splitter.split.Unsplitter(
+            all_stacks=True,
+            backend_dir=self._tmp_dir,
+        ):
+            got = util.Directory.load(self._tmp_dir)
+            want = data.multi_stack_unsplit()
+            want.compare(got, self)
+
+        got = util.Directory.load(self._tmp_dir)
+        input_.compare(got, self)
+
+    def test_unsplitter_new(self):
+        """Testing `Unsplitter` with with a new stack added."""
```

Comment:
```suggestion
        """Testing `Unsplitter` with a new stack added."""
```

Reply:
Done in 7341de463e7e67935f8aafb6d238d35ed4bcf16d .